kernel/rtlil.h

```diff
@@ -968,15 +969,16 @@ struct RTLIL::Module : public RTLIL::AttrObject
 	RTLIL::Design *design;
 	pool<RTLIL::Monitor*> monitors;
 
+	int refcount_parameters_;
 	int refcount_wires_;
 	int refcount_cells_;
 
+	dict<RTLIL::IdString, RTLIL::Parameter*> parameters_;
 	dict<RTLIL::IdString, RTLIL::Wire*> wires_;
 	dict<RTLIL::IdString, RTLIL::Cell*> cells_;
 	std::vector<RTLIL::SigSig> connections_;
 
 	RTLIL::IdString name;
-	pool<RTLIL::IdString> avail_parameters;
```

Comment:
In order to preverse the fast path, it probably makes sense to maintain both avail_parameters **and** the parameters_ dict.

Reply:
But then we would have two structures which store the redundant information (parameter name). Not to mention that inserting data into two structures simultaneously will be more time+mem consuming than into only one.

If we really want it optimal than it would be better to have the avail_parameters as it was and add another structure which will hold attributes, but only for those parameters which have them.

Reply:
> If we really want it optimal than it would be better to have the avail_parameters as it was and add another structure which will hold attributes, but only for those parameters which have them.

That's more or less what I'm thinking.

Reply:
Done.